internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -329,6 +329,15 @@ func (f *FleetGateway) execute(ctx context.Context) (*fleetapi.CheckinResponse,
 	// convert components into checkin components structure
 	components := f.convertToCheckinComponents(state.Components)
 
+	if ecsMeta.Elastic == nil || ecsMeta.Elastic.Agent == nil {
+		// escMeta struct is incomplete: log a warning
+		f.log.Warnw("Agent ECSMetadata struct is missing/incomplete", "elastic_ecs_metadata", ecsMeta.Elastic)
```

Comment:
It can, the fact is that the current implementation of `AgentInfo.ECSMetadata()` guarantees that those structures exist but they are pointers so I would like to avoid a segmentation fault when something changes...
Maybe I can replace that with a unit test + assertion that `AgentInfo.ECSMetadata()` always returns initialized `Elastic` and `Elastic.Agent` pointers

Reply:
removed in 49738443f6ec55ca69bf7e36f0b7fe42a1e7d39b in favor of assertions on `ECSMetadata()` unit tests